File: core/src/main/java/net/dontdrinkandroot/wicket/converter/LocalDateTimeConverter.java
package net.dontdrinkandroot.wicket.converter;

import java.time.LocalDateTime;

/**
 * Converts between String and {@link java.time.LocalDateTime}
 *
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class LocalDateTimeConverter extends AbstractTemporalAccessorConverter<LocalDateTime>
{
    public LocalDateTimeConverter()
    {
        super("yyyy-MM-dd'T'HH:mm", LocalDateTime::from);
    }
}

File: core/src/main/kotlin/net/dontdrinkandroot/wicket/converter/LocalDateTimeConverter.kt
package net.dontdrinkandroot.wicket.converter

import java.time.LocalDateTime
import java.time.temporal.TemporalAccessor
import java.time.temporal.TemporalQuery

/**
 * Converts between String and [java.time.LocalDateTime]
 */
class LocalDateTimeConverter : AbstractTemporalAccessorConverter<LocalDateTime>(
    "yyyy-MM-dd'T'HH:mm",
    TemporalQuery { temporal: TemporalAccessor -> LocalDateTime.from(temporal) })